activerecord/test/cases/migration/check_constraint_test.rb

```diff
@@ -44,6 +54,18 @@ def test_check_constraints
             assert_equal "price > discounted_price", constraint.expression
           end
 
+          if current_adapter?(:Mysql2Adapter)
```

Comment:
Code looks good but this new test is flaky https://buildkite.com/rails/rails/builds/95406#01874c8b-219b-4629-8c29-d7070ffbf1cf/1066-1075 (didn't fail on all mysql builds but did fail on some).

Reply:
With my last change, it succeeds everywhere EXCEPT for MariaDB... which looks like it might be because the `JSON_SCHEMA_VALID` function only exists in MySQL 8.0. I'm not sure if we need to care about that or not; I'm inclined to switch to another function that definitely does exist.

Reply:
I've moved to `JSON_CONTAINS`, which looks like it's fine on both MySQL and MariaDB. It wasn't important in this case which function call was being used, just that it was correctly handled when it _was_ used.